pkg/server/config.go

```diff
@@ -224,12 +225,18 @@ func NewConfig(opts *kcpserveroptions.CompletedOptions) (*Config, error) {
 		}
 		c.identityConfig, c.resolveIdentities = boostrap.NewConfigWithWildcardIdentities(c.GenericConfig.LoopbackClientConfig, boostrap.KcpRootGroupExportNames, boostrap.KcpRootGroupResourceExportNames, nonIdentityKcpClusterClient, nil)
 	}
-	c.KcpClusterClient, err = kcpclient.NewClusterForConfig(c.identityConfig) // this is now generic to be used for all kcp API groups
+	c.KcpClusterClient, err = kcpclient.NewForConfig(kcpclienthelper.NewClusterConfig(c.identityConfig)) // this is now generic to be used for all kcp API groups
```

Comment:
We've changed `NewClusterConfig` to not do a copy of the `*rest.Config`, so we will want to explicitly do that when calling the new helper.